[Health check] Add logs with the details about the check

Desvelao opened the pull request:
### Description
This PR adds logs with the details about the checks.
- Added button to open/close the check details
- Added a new section in Settings: `Miscellaneous`:
  - Health check: run in `debug` mode
- Refactorized how the check validator function is managed
- Refactorized another function
- Some adjustments in Settings section related to spaces

### Checks
- Health check:
  - Open/close the check details with the button added after the check result
  - When a check does an `action`, the button to open the details should have a notification icon
  
- Settings > Miscellaneous:
  - The new section should be visible and accessible in the Settings section
    - Try the link in the main menu
    - Try the tab in Settings
  - Run the Health check in `debug` mode:
    - Should redirect to health check
    - A `Continue` button should appear when all checks are ready
    
- Considers reviewing the messages displayed in the logs
- Try with index patterns not created

### Suggested improvements
- [x] Remove the notification after the check logs are opened

Closes: #3208

CPAlejandro reviewed (changes requested):
- When stopping the manager service and executing the health-check, an error of an uncontrolled undefined appears.
![image](https://user-images.githubusercontent.com/17100196/118796799-3ba4d800-b89c-11eb-811a-d6ad7649c2db.png)

- As a suggestion, maybe when a red dot appears and you open the logs, it could disappear?

Desvelao commented:
I like the suggestion to remove the notification after opening the check log. I will wait until somebody reviews the code and tests it

Desvelao commented:
> * When stopping the manager service and executing the health-check, an error of an uncontrolled undefined appears.
>   ![image](https://user-images.githubusercontent.com/17100196/118796799-3ba4d800-b89c-11eb-811a-d6ad7649c2db.png)

I tried to replicate the problem and I can't get the same error displayed in the screenshot. Could someone try to replicate?

I tried:
- Stop the `wazuh-manager` service
```service wazuh-manager stop```
- Turn off the machine with the manager (in my environment using Docker container, I stoped the container)

CPAlejandro reviewed (commented):
- Could we control the use of toast to not be able to display the same one twice, which can be annoying?
![image (3)](https://user-images.githubusercontent.com/17100196/119326035-c8c3a480-bc81-11eb-8b8f-de15c875ee36.png)
- When moving from branch 4.2-7.10 to this branch and executing the health-check from / settings / Miscellaneous I get the following error
x-special/nautilus-clipboard
copy
![image (4)](https://user-images.githubusercontent.com/17100196/119326252-fd376080-bc81-11eb-8d13-3049dc84df23.png)
![image (5)](https://user-images.githubusercontent.com/17100196/119326276-03c5d800-bc82-11eb-84b5-dd032df5a7d0.png)

Desvelao commented:
### Changes
- Fix unknown error after restarting the cluster
- Removed the notification dot when opening for the first time the check details
- Updated tests

CPAlejandro reviewed (commented):
Testing: LGTM!

pablomarga reviewed (approved):
CR: :heavy_check_mark: 
Testing: :heavy_check_mark:

Desvelao commented:
Someone to review the log messages @wazuh/frontend .

asteriscos commented:
Team,
made some small improvements to the UI: 

- Fixed some verbose log texts
- Removed redundant status texts
- Improved the styling

A quick review is needed please!

![healthcheck](https://user-images.githubusercontent.com/9343732/119542827-5bd60a80-bd66-11eb-823a-41e06dd686fd.gif)

asteriscos commented on public/components/health-check/components/check-result.tsx:
```diff
@@ -112,21 +150,73 @@ export function CheckResult(props) {
+      content={tooltipVerboseButton}
+      >
+      <EuiButtonEmpty size='xs' onClick={switchVerboseDetails} isDisabled={!haveLogs} textProps={{ style:{overflow: 'visible' , position: 'relative'}}}>
+        <EuiButtonIcon
```
This button inception could be avoided?

frankeros commented on public/components/health-check/components/check-result.tsx:
```diff
@@ -112,21 +150,73 @@ export function CheckResult(props) {
+    setVerboseIsOpen(state => !state);
+  },[checkDidSomeAction]);
+
+  const inspectLogsButton = isCheckStarted && haveLogs ? 
```
pls move the button and logs component to separate child components in order to keep a short class

asteriscos replied:
Great idea, already split the components. Please review them!

frankeros reviewed (changes requested):
Test: :heavy_check_mark: 
CR: changes requestd

I tested and it looks good. But I think that is a little confusing that the user gets the chance to open the logs then the health-check redirects automatically, 
I would like that the button and logs to be displayed only when there is an error in normal mode. 
In verbose mode it's ok.

asteriscos commented:
> Test: 
> CR: changes requestd
> 
> I tested and it looks good. But I think that is a little confusing that the user gets the chance to open the logs then the health-check redirects automatically,
> I would like that the button and logs to be displayed only when there is an error in normal mode.
> In verbose mode it's ok.

Sounds good. Now the log buttons will show when there's an error or in verbose mode. Please review it!

CPAlejandro reviewed (approved):
Testing: LGTM!

frankeros reviewed (approved):
Test: :heavy_check_mark: 
CR: :heavy_check_mark: 
Amazing job guys!! :tada:
